contrib/babelfishpg_tsql/sql/ownership.sql

```diff
@@ -366,10 +366,33 @@ CAST(CASE WHEN Ext.type = 'R' THEN 1 ELSE Ext.owning_principal_id END AS INT) AS
 CAST(CASE WHEN Ext.type = 'R' THEN 1 ELSE Ext.is_fixed_role END AS sys.BIT) AS is_fixed_role
 FROM pg_catalog.pg_roles AS Base INNER JOIN sys.babelfish_authid_login_ext AS Ext ON Base.rolname = Ext.rolname
 WHERE (pg_has_role(suser_id(), 'sysadmin'::TEXT, 'MEMBER')
-OR Ext.orig_loginname = suser_name()
-OR Ext.orig_loginname = (SELECT pg_get_userbyid(datdba) FROM pg_database WHERE datname = CURRENT_DATABASE()) COLLATE sys.database_default
-OR Ext.type = 'R')
-AND Ext.type != 'Z';
+  OR Ext.orig_loginname = suser_name()
+  OR Ext.orig_loginname = (SELECT pg_get_userbyid(datdba) FROM pg_database WHERE datname = CURRENT_DATABASE()) COLLATE sys.database_default
+  OR Ext.type = 'R')
+  AND Ext.type != 'Z'
+UNION ALL
+SELECT
+CAST(name AS SYS.SYSNAME) AS name,
+CAST(2 AS INT) AS principal_id,
```

Comment:
Does it exist internally on the PG side?

Reply:
No it doesn't. This is the following output we get when we query the names in the pg_catalog.pg_roles
```
1> SELECT rolname FROM pg_catalog.pg_roles
2> go
rolname                                                                                                                                                                                                                                                         
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
aswinii                                                                                                                                                                                                                                                         
pg_database_owner                                                                                                                                                                                                                                               
pg_read_all_data                                                                                                                                                                                                                                                
pg_write_all_data                                                                                                                                                                                                                                               
pg_monitor                                                                                                                                                                                                                                                      
pg_read_all_settings                                                                                                                                                                                                                                            
pg_read_all_stats                                                                                                                                                                                                                                               
pg_stat_scan_tables                                                                                                                                                                                                                                             
pg_read_server_files                                                                                                                                                                                                                                            
pg_write_server_files                                                                                                                                                                                                                                           
pg_execute_server_program                                                                                                                                                                                                                                       
pg_signal_backend                                                                                                                                                                                                                                               
pg_checkpoint                                                                                                                                                                                                                                                   
pg_use_reserved_connections                                                                                                                                                                                                                                     
pg_create_subscription                                                                                                                                                                                                                                          
jdbc_user                                                                                                                                                                                                                                                       
bbf_role_admin                                                                                                                                                                                                                                                  
sysadmin                                                                                                                                                                                                                                                        
master_db_owner                                                                                                                                                                                                                                                 
master_dbo                                                                                                                                                                                                                                                      
master_guest                                                                                                                                                                                                                                                    
tempdb_db_owner                                                                                                                                                                                                                                                 
tempdb_dbo                                                                                                                                                                                                                                                      
tempdb_guest                                                                                                                                                                                                                                                    
msdb_db_owner                                                                                                                                                                                                                                                   
msdb_dbo                                                                                                                                                                                                                                                        
msdb_guest                                                                                                                                                                                                                                                      
testlogin                                                                                                                                                                                                                                                       

(28 rows affected)
```